src/PIL/Image.py

```diff
@@ -2927,14 +2939,17 @@ def new(mode, size, color=0):
 
         color = ImageColor.getcolor(color, mode)
 
-    im = Image()
+    rgb_color = None
     if mode == "P" and isinstance(color, (list, tuple)) and len(color) in [3, 4]:
         # RGB or RGBA value for a P image
-        from . import ImagePalette
+        rgb_color = color
 
-        im.palette = ImagePalette.ImagePalette()
-        color = im.palette.getcolor(color)
-    return im._new(core.fill(mode, size, color))
+        # This will be the first color allocated to the palette
```

Comment:
Why we need to make this assumption rather than just using existing API?

We can rewrite palette if it is exist:

```python
im = Image._init(core.fill(mode, size, color))
if palette:
     im.palette = palette
```

Reply:
Hmm, I understand the desire to use APIs. My original thought was to try and avoid constructing a second imagePalette, but this version of the code seems simpler to me. I don't think it should ever change that a palette starts out empty, and that the first color is at the zero index.

Are you sure using `getcolor()` is better?

Reply:
@homm did my argument persuade you, or no? I can change to your suggestion if you like, I'd rather progress this than let it sit unresolved.